### PR TITLE
Fix Reply layout

### PR DESCRIPTION
**Pull request by mplorentz**
This fixes the issues described [here](https://github.com/planetary-social/nos/issues/450#issuecomment-2018317754) with the layout of the new reply UI.

I spent a long time fighting the layout system on this one, and in the end I settled for a compromise. The main difficulty was in getting the text view to expand to fill available space when there isn't so much content that you need to scroll to see it all. We need to do this because people try to tap (i.e. to paste) in the space below the text view and nothing happens, it's a bug we've fixed in the past. This was hard because `GeometryReader` is all sorts of broken when it is inside a `ScrollView` which meant I couldn't just do the math to see if there was available space and change the minHeight on the editor accordingly. 

What I did instead was to just make the minimumHeight on the text view the height of the `ScrollView` frame. This means that in a lot of cases it is too big, and there is an awkward extra space you can scroll down to. I changed the logic that scrolls the text view into view so that it doesn't scroll all the way to the bottom, and I think it will be barely noticeable to the user.

I also renamed `EditableText` because it's name was confusing alonside `NoteTextEditor` and `EditableNoteText`.

### CODE REVIEW

File: Nos/Views/New Note/NewNoteView.swift [hunk 1]
```diff
@@ -40,37 +43,50 @@ struct NewNoteView: View {
         self.replyToNote = replyTo
     }
     
+    // We do this because editor won't expand to fill available space when it's in a ScrollView.
+    // and we need it to because people try to tap below the text field bounds to paste if it doesn't
+    // fill the screen.
```
[the remark this answers is not in the document]
Comment (mplorentz) [reply Nos/Views/New Note/NewNoteView.swift:48]: yes thank you. It got copy-pasted from inline somewhere. I have updated it.

File: Nos/Views/NoteTextViewRepresentable.swift [hunk 2]
```diff
@@ -2,16 +2,21 @@ import Foundation
...
 /// to nostr objects when it is received.
-struct EditableText: UIViewRepresentable {
+struct NoteTextViewRepresentable: UIViewRepresentable {
 
     typealias UIViewType = UITextView
 
     @Binding var text: EditableNoteText
+    
+    /// The height that fits all entered text. This value will be updated by EditableText automatically, and should
+    /// be used to set the frame of EditableText from SwiftUI. This is done to work around some incompatibilities
```
[the remark this answers is not in the document]
Comment (mplorentz) [reply Nos/Views/NoteTextViewRepresentable.swift:17]: yes

File: Nos/Views/NoteTextViewRepresentable.swift [hunk 3]
```diff
@@ -167,9 +191,10 @@ extension Notification.Name {
 struct EditableText_Previews: PreviewProvider {
```
[the remark this answers is not in the document]
Comment (mplorentz) [reply Nos/Views/NoteTextViewRepresentable.swift:191]: I just changed the name.

**Review by joshuatbrown — APPROVED** [review]
🚀

**Pull request merged**